src/main/Application.cpp

```diff
@@ -45,30 +45,58 @@ Application::create(VirtualClock& clock, Config const& cfg, bool newDB,
 }
 
 bool
-Application::modeHasOverlay(AppMode m)
+Application::modeUsesInMemoryDB(AppMode m)
```

Comment:
I think `AppMode` should be removed:
* we should control modes via the `Config` class
* app modes are really "shorthands" (exposed to users) that allow specific combinations of "switches", like `MODE_BUCKET_LIST_ENABLED`. So that we can control which combinations of switches are valid.